### PR TITLE
interpreter: support calling function with the invoke function

### PR DESCRIPTION
**Pull request by ogoffart**
Deprecate the invoke_callback one

CC: #2012

### CODE REVIEW

File: internal/interpreter/api.rs [hunk 1]
```diff
@@ -967,24 +971,51 @@ impl ComponentInstance {
...
+            comp.description()
+                .original
+                .root_element
+                .borrow()
+                .lookup_property(callable_name)
+                .property_type,
+            LangType::Function { .. }
+        ) {
+            g.as_ref()
+                .eval_function(callable_name, args.iter().cloned().collect())
```
Comment (tronical) [inline internal/interpreter/api.rs:1012]: Oh there's no need to normalize for functions?

File: internal/interpreter/api.rs [hunk 2]
```diff
@@ -1087,13 +1118,22 @@ pub enum SetCallbackError {
...
 #[derive(Debug, Clone, Copy, PartialEq, Eq, thiserror::Error)]
 #[non_exhaustive]
-pub enum InvokeCallbackError {
-    /// There is no callback with the given name
-    #[error("no such callback")]
-    NoSuchCallback,
+pub enum InvokeError {
+    /// There is no callback or function with the given name
+    #[error("no such callback or function")]
+    NoSuchCallable,
```
Comment (tronical) [inline internal/interpreter/api.rs:1127]: 👍

File: internal/interpreter/ffi.rs [hunk 3]
```diff
@@ -507,28 +507,41 @@ pub unsafe extern "C" fn slint_interpreter_component_instance_set_global_callbac
...
+                comp.description()
+                    .original
+                    .root_element
+                    .borrow()
+                    .lookup_property(callable_name)
+                    .property_type,
+                i_slint_compiler::langtype::Type::Function { .. }
+            ) {
+                g.as_ref().eval_function(
+                    &normalize_identifier(callable_name),
```
Comment (tronical) [inline internal/interpreter/ffi.rs:539]: Ah here the name is normalized. I think earlier it would need to be as well?

**Review by tronical — APPROVED** [review]

File: internal/interpreter/api.rs (hunk 1 above)
Comment (ogoffart) [reply internal/interpreter/api.rs:1012]: well spotted!

**Review by tronical — APPROVED** [review]

**Pull request merged**